src/deploy/functions/build.ts

```diff
@@ -0,0 +1,397 @@
+    }
+    if (endpoint.eventTrigger.region) {
+      bkEvent.region = resolveString(endpoint.eventTrigger.region);
+    }
+    trigger = { eventTrigger: bkEvent };
+  } else if ("scheduleTrigger" in endpoint) {
+    const bkSchedule: backend.ScheduleTrigger = {
+      schedule: endpoint.scheduleTrigger.schedule,
+      timeZone: endpoint.scheduleTrigger.timeZone,
+    };
+    proto.renameIfPresent(
+      bkSchedule,
+      endpoint.scheduleTrigger,
+      "retryConfig",
+      "retryConfig",
+      resolveInt
+    );
+    trigger = { scheduleTrigger: bkSchedule };
+  } else if ("taskQueueTrigger" in endpoint) {
+    const bkTaskQueue: backend.TaskQueueTrigger = {};
+    if (endpoint.taskQueueTrigger.rateLimits) {
+      const bkRateLimits: backend.TaskQueueRateLimits = {};
+      proto.renameIfPresent(
+        bkRateLimits,
+        endpoint.taskQueueTrigger.rateLimits,
+        "maxConcurrentDispatches",
+        "maxConcurrentDispatches",
+        resolveInt
+      );
+      proto.renameIfPresent(
+        bkRateLimits,
+        endpoint.taskQueueTrigger.rateLimits,
+        "maxDispatchesPerSecond",
+        "maxDispatchesPerSecond",
+        resolveInt
+      );
+      bkTaskQueue.rateLimits = bkRateLimits;
+    }
+    if (endpoint.taskQueueTrigger.retryConfig) {
+      const bkRetryConfig: backend.TaskQueueRetryConfig = {};
+      proto.renameIfPresent(
+        bkRetryConfig,
+        endpoint.taskQueueTrigger.retryConfig,
+        "maxAttempts",
+        "maxAttempts",
+        resolveInt
+      );
+      proto.renameIfPresent(
+        bkRetryConfig,
+        endpoint.taskQueueTrigger.retryConfig,
+        "maxBackoffSeconds",
+        "maxBackoffSeconds",
+        (from: number | Expression<number> | null): string => {
+          return proto.durationFromSeconds(resolveInt(from));
+        }
+      );
+      proto.renameIfPresent(
+        bkRetryConfig,
+        endpoint.taskQueueTrigger.retryConfig,
+        "minBackoffSeconds",
+        "minBackoffSeconds",
+        (from: number | Expression<number> | null): string => {
+          return proto.durationFromSeconds(resolveInt(from));
+        }
+      );
+      proto.renameIfPresent(
+        bkRetryConfig,
+        endpoint.taskQueueTrigger.retryConfig,
+        "maxRetrySeconds",
+        "maxRetryDurationSeconds",
+        (from: number | Expression<number> | null): string => {
+          return proto.durationFromSeconds(resolveInt(from));
+        }
+      );
+      proto.renameIfPresent(
+        bkRetryConfig,
+        endpoint.taskQueueTrigger.retryConfig,
+        "maxDoublings",
+        "maxDoublings",
+        resolveInt
+      );
+      bkTaskQueue.retryConfig = bkRetryConfig;
+    }
+    if (endpoint.taskQueueTrigger.invoker) {
+      bkTaskQueue.invoker = endpoint.taskQueueTrigger.invoker.map((sa) => resolveString(sa));
+    }
+    trigger = { taskQueueTrigger: bkTaskQueue };
+  } else {
+    throw new FirebaseError("unknown trigger type");
```

Comment:
If you've done the typing right you should be able to just say `assertExhaustive(endpoint)` and it'll fail to compile if we ever add a new trigger type.